### PR TITLE
[Fixes #1878] Comparing xml content in tests

### LINKED ISSUE #1878: Investigate Xml formatters tests failures on Windows 2008 R2 OS
image

### PR DESCRIPTION
**Pull request by kichalla**
@yishaigalatzer @rynowak

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using System.Linq;
+using System.Xml.Linq;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNet.Mvc.FunctionalTests
+{
+    /// <summary>
+    /// Xunit assertions related to Xml content.
+    /// </summary>
+    public static class XmlAssert
```
Comment (kichalla) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs]: This code has been duplicated in this project as well as in the `Microsoft.AspNet.Mvc.Xml.Test`

**Comment by kichalla** [comment]
@harshgMSFT @sornaks

**Comment by kichalla** [comment]
ping

File: test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs (hunk 1 above)
Comment (yishaigalatzer) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs]: put it in test common

File: test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+            bool areEqual = XNode.DeepEquals(sortedExpectedXDoc, sortedActualXDoc);
+
+            if (!areEqual)
+            {
+                throw new EqualException(expectedXml, actualXml);
+            }
+        }
+
+        private static XDocument Sort(XDocument doc)
```
Comment (yishaigalatzer) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs]: this should also be sort attributes

**Comment by yishaigalatzer** [comment]
:shipit: after making the requested changes

**Comment by kichalla** [comment]
Commit 2e938a51f86c4d1868341079c195a336819adb53

File: test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs (hunk 1 above)
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/XmlAssert.cs]: @kichalla new `../Microsoft.AspNet.Mvc.Xml.Test/XmlAssert.cs` reference in the functional tests project is a bit strange.  why isn't this class in a new `Microsoft.AspNet.Mvc.Tests.Common` project or perhaps in the existing `Microsoft.AspNet.Testing` project (in testing repo)?

**Pull request closed without merging**